leetduel-backend/src/main.py

```diff
@@ -216,11 +288,31 @@ async def submit_code(sid: str, data: dict) -> None:
         await sio.emit("player_submit", {"message": message_to_room, "bold": True, "color": color}, room=party_code)
 
     if all_players_passed(party_code):
-        parties[party_code]["status"] = "waiting"
-        reset_players_passed(party_code)
-        await sio.emit("announcement", {"message": "All players passed! Game over."}, room=party_code)
-        await asyncio.sleep(3)
-        await sio.emit("game_over", room=party_code)
+        party = parties[party_code]
+        for player in party["players"]:
+            player["total_score"] += player.get("current_score", 0)
+        leaderboard = sorted(party["players"], key=lambda x: x["total_score"], reverse=True)
+        await sio.emit(
+            "round_leaderboard",
+            {
+                "leaderboard": [{"username": p["username"], "score": p["total_score"]} for p in leaderboard],
+                "round": party["current_round"],
+                "total_rounds": party["total_rounds"],
```

Comment:
nit: also nice to have this typed so it's easy for future collaborators to make frontend/backend consistent